### PR TITLE
Tests if an input is valid PHP serialized string

### PR DESCRIPTION
**Pull request by RLovelett**
A quick and dirty test to see if a string can even be unserialized.

### CODE REVIEW

File: lib/php_serialize.rb [hunk 1]
```diff
@@ -210,6 +210,40 @@ def string.read_until(char)
...
+  # Checks if a string is serialized using quick string manipulation
+  # to throw out obviously incorrect strings.
+  #
+  # http://stackoverflow.com/questions/1369936/check-to-see-if-a-string-is-serialized
+  def PHP.serialized?(string)
+    # If it isn't a string, it isn't serialized
+    return false unless string.instance_of? String
+
+    # Remove any trailing whitespace
+    string.chomp!
```
Comment (jqr) [inline lib/php_serialize.rb]: Mutating the argument is a bit strange, seems like we should make a new variable here. Or alternatively, should we be chomping at all? If a end-user adds whitespace, it's up to them to delete it.

**Comment by Freaky** [comment]
Is there a reasonable use-case for this?  Don't you know what your data is supposed to be? :/

**Comment by RLovelett** [comment]
TBH I have no idea what this is anymore. I've long since stopped writing PHP. Considering I made this PR in 2012, +6 years ago, we can safely assume I've worked around the issue. If you don't like the code we can just close the PR.

**Comment by Freaky** [comment]
Thanks @RLovelett, sorry we weren't really there for you before, and glad you're in a better place now :)

**Pull request closed without merging**